extensions/interactions/rules.json

```diff
@@ -0,0 +1,186 @@
+{
+  "CodeRepl": {
```

Comment:
It's a good question -- I'm a bit torn. On the one hand it's nice that they are self-contained within their interactions, and I can see the attraction of that. On the other hand, it's simpler and cheaper for the backend/frontend to read from a single file -- kind of like feconf.py. (Ultimately we'll probably refer to this file from the frontend as well, for getting the rule definitions.) It's also easier for the Karma/Protractor tests to reference a single rules.json, since it's a bit trickier to run a for loop over json imports.

For now, I lean towards keeping everything in a single file, and writing Karma tests to ensure that the rule evaluation definitions for each interaction coincide with what's in this file (which is the approach I've taken in this commit), so that any skew between the specifications will be quickly noticed and corrected. But, I'm not entirely sure. @BenHenning, @wxyxinyu, what do you think?